frame/assets/src/lib.rs

```diff
@@ -468,26 +487,14 @@ decl_module! {
 			#[compact] amount: T::Balance
 		) -> DispatchResult {
 			let origin = ensure_signed(origin)?;
-			let beneficiary = T::Lookup::lookup(beneficiary)?;
 
-			Asset::<T>::try_mutate(id, |maybe_details| {
-				let details = maybe_details.as_mut().ok_or(Error::<T>::Unknown)?;
+			ensure!(T::AllowMinting::get(), Error::<T>::NoMintingAllowed);
```

Comment:
not sure if we want to do feature flag check here or inside `<Self as MintableAsset<_>>::mint`

Reply:
This is what @shawntabrizi said we should not do.

Reply:
You can see that the low level APIs do not have any of these feature flag stuff. You can import this module without any `Call` and use it simple as the backend for any token system you desire. As such, your interaction layer should be the one to check if minting is allowed, or whether you even want to constrain to this trait when using this Asset.

But would be interesting to see if you disagree here.

Reply:
I guess the ambiguity lies in the purpose the flag, does it protect only the extrinsics or protect the whole pallet.
Improved docs should make this clear it only protect the extrinsics.

Reply:
Sure, documentation will help here.

But here is an example why the flag i think could never apply at the whole pallet. Imagine some other pallet which needs an implementer of `FungibleAsset + MintableAsset + BurnableAsset`. Let's say the functionality of this pallet depends on these traits. If we then allowed the pallet to disable any of these functions at the pallet level, Rust will allow this pallet to satisfy those trait conditions, but we would know that it wouldn't work.

I guess it would be that for this kind of scenario, feature flags could allow us to include certain trait implementations dynamically, but as I mentioned, feature flags are a little bit scary to me for the runtime.

Reply:
Cargo feature will make sure you get compile time error. But it have its own problem.